src/arrayMethodSplice.js

```diff
@@ -5,7 +5,78 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const deleteCountSplice = deleteCount;
+    const arrayItems = [...items];
+    const argumentsLength = arguments.length;
+    const sourceLength = this.length;
+    const returnedArray = [];
+    let startOfSplice = start;
+    // assign all constants for next work with them
+
+    // first situation with empty splice
+    if (argumentsLength === 0) return returnedArray;
+
+    // change start of splice if start < 0
+    if (startOfSplice < 0) {
+      startOfSplice += sourceLength;
+      if (startOfSplice < 0) { // checking if start is still less then 0
+        startOfSplice = Math.abs(startOfSplice);
+      }
+      if (startOfSplice > this.length) startOfSplice = 0;
```

Comment:
Why this if is inside `if (startOfSplice < 0) {` it cannot be greater then `this.length`
be consistent use `sourceLength `